### PR TITLE
Add `UpdateUser` `UpdateUserPassword` `DeleteUser`

### PR DESCRIPTION
**Pull request by pliao28**
## Description
Add `UpdateUser` `UpdateUserPassword` `DeleteUser`
*Fixes USRLD-961, USRLD-948, USRLD-955
## Documentation

Does this require changes to the WorkOS Docs? E.g. the [API Reference](https://workos.com/docs/reference) or code snippets need updates.

```
[ ] Yes
```

If yes, link a related docs PR and add a docs maintainer as a reviewer. Their approval is required.

### CODE REVIEW

File: src/WorkOS.net/Services/UserManagement/_interfaces/UpdateUserPasswordOptions.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+{
+    using Newtonsoft.Json;
+
+    /// <summary>
+    /// The parameters to update the password of the user.
+    /// </summary>
+    public class UpdateUserPasswordOptions : BaseOptions
+    {
+        /// <summary>
+        /// The id of the user.
```
Comment (maxdeviant) [inline src/WorkOS.net/Services/UserManagement/_interfaces/UpdateUserPasswordOptions.cs:11]:
```suggestion
        /// The ID of the user.
```

File: src/WorkOS.net/Services/UserManagement/_interfaces/UpdateUserPasswordOptions.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        /// </summary>
+        public string Id { get; set; }
+
+        /// <summary>
+        /// The email address of the user.
```
Comment (maxdeviant) [inline src/WorkOS.net/Services/UserManagement/_interfaces/UpdateUserPasswordOptions.cs:16]:
```suggestion
        /// The new password for the user.
```

File: src/WorkOS.net/Services/UserManagement/_interfaces/AuthenticateUserWithMagicAuthOptions.cs [hunk 3]
```diff
@@ -32,10 +32,10 @@ public class AuthenticateUserWithMagicAuthOptions : BaseOptions
         public string Code { get; set; }
 
         /// <summary>
-        /// The challenge ID returned from when the one-time code was sent to the user.
+        /// The Id of User authenticating.
```
Comment (maxdeviant) [inline src/WorkOS.net/Services/UserManagement/_interfaces/AuthenticateUserWithMagicAuthOptions.cs:35]:
```suggestion
        /// The ID of the User authenticating.
```

File: src/WorkOS.net/Services/UserManagement/UserManagementService.cs [hunk 4]
```diff
@@ -259,5 +259,61 @@ public UserManagementService(WorkOSClient client)
...
+        /// <returns> The corresponding User object.</returns>
+        public async Task<User> UpdateUser(
+            UpdateUserOptions options,
+            CancellationToken cancellationToken = default)
+        {
+            var request = new WorkOSRequest
+            {
+                Options = options,
+                Method = HttpMethod.Put,
+                Path = $"/users/{options.Id}",
```
Comment (pliao28) [inline src/WorkOS.net/Services/UserManagement/UserManagementService.cs:277]: @maxdeviant  Getting your opinion on new way of passing `id`, think it makes more sense to pass id through the options interface than having 2 seperate params in `id` and `options`, if this doesn't make sense I can revert back to previous method. If it makes sense I'll pull another PR to fix across the SDK

Comment (maxdeviant) [reply src/WorkOS.net/Services/UserManagement/UserManagementService.cs:277]: One difference with this approach is that now `id` will get passed on `Options` as well in the serialized form, so we'll have to make sure to explicitly mark it to be excluded from serialization.

Comment (pliao28) [reply src/WorkOS.net/Services/UserManagement/UserManagementService.cs:277]: Right, think we should be able to just add [JsonIgnore] to those fields to ignore serialization then.

File: src/WorkOS.net/Services/UserManagement/_interfaces/AuthenticateUserWithMagicAuthOptions.cs (hunk 3 above)
Comment (pliao28) [reply src/WorkOS.net/Services/UserManagement/_interfaces/AuthenticateUserWithMagicAuthOptions.cs:35]: Doing this change on #149

**Review by maxdeviant — APPROVED** [review]

**Pull request merged**